src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -48,37 +59,245 @@ public void run(Connection conn, List<Object> params) throws SQLException {
+                                break;
+                            case SCRAMBLED:
+                                ScrambledZipfianGenerator scramGen = (ScrambledZipfianGenerator) param.getGenerator();
+                                stmt.setTimestamp(i + 1, new Timestamp(scramGen.nextLong()));
+                                break;
+                            default:
+                                throw new RuntimeException(
+                                        "Distribution: " + distribution + " not supported for type: " + paramType);
+                        }
+                        break;
+                    case "date":
+                        Long dateMin = param.getMin();
+                        Long dateMax = param.getMax();
+                        switch (distribution) {
+                            case UNIFORM:
+                                Random uniformRand = (Random) param.getGenerator();
+                                stmt.setDate(i + 1, new Date(uniformRand.nextLong(dateMin, dateMax)));
+                                break;
+                            case BINOMIAL:
+                                Random binomialRandom = (Random) param.getGenerator();
+                                Long lVal;
+                                do {
+                                    lVal = Double.valueOf(dateMin + Math.abs(binomialRandom.nextGaussian()) * dateMax)
+                                            .longValue();
+                                } while (lVal > dateMax || lVal < dateMin);
+
+                                stmt.setDate(i + 1, new Date(lVal));
+                                break;
+                            case ZIPFIAN:
+                                ZipfianGenerator zipfGen = (ZipfianGenerator) param.getGenerator();
+                                stmt.setDate(i + 1, new Date(zipfGen.nextLong()));
+                                break;
+                            case SCRAMBLED:
+                                ScrambledZipfianGenerator scramGen = (ScrambledZipfianGenerator) param.getGenerator();
+                                stmt.setDate(i + 1, new Date(scramGen.nextLong()));
+                                break;
+                            default:
+                                throw new RuntimeException(
+                                        "Distribution: " + distribution + " not supported for type: " + paramType);
+                        }
+                        break;
+                    case "time":
+                        Long timeMin = param.getMin();
+                        Long timeMax = param.getMax();
+                        switch (distribution) {
+                            case UNIFORM:
+                                Random uniformRand = (Random) param.getGenerator();
+                                stmt.setTime(i + 1, new Time(uniformRand.nextLong(timeMin, timeMax)));
+                                break;
```

Comment:
There is a lot of repetition in many of these blocks.

For starters, in each block, the `stmt.setSomeType(i + 1, generatedValue)` portion is repeated.  It could just go at the end of the internal switch block.

But once you do that, I think you'll find that many of the internal distribution switch statements are repeated across many of the types.
For instance, most of them just look up and cast the appropriate distribution type, then return a value within the specified range using a `long`.  I think `ints` could be handled using the `long` code and just throwing an exception either on construction time when the range is first given (best option since the error code only happens once and early then) or at runtime when the `long` value generated is not castable into the `int` range.

That leaves floats and strings as the only exceptions.

Anyways, I think all of that `genNextValue` could probably be abstracted into the `TemplatedValue` class and called at the top of this block here so that mostly the only thing this switch statement is casting the generated value to the appropriate sql type when setting the parameter.